core-dictionary/src/main/java/org/apache/kylin/dict/TrieDictionaryBuilder.java

```diff
@@ -152,10 +158,10 @@ else if (comp > 0)
         }
         if (found) {
             // found a child node matching the first byte, continue in that child
-            addValueR(node.children.get(mid), value, j);
+            addValueR(node.children.get(mid), value, j, isSplitValue);
         } else {
             // otherwise, make the value a new child
-            Node c = new Node(BytesUtil.subarray(value, j, nn), true);
+            Node c = new Node(BytesUtil.subarray(value, j, nn), isSplitValue ? false : true);
```

Comment:
`isSplitValue? false : true` can be simplified to `!isSplitValue`

Reply:
thanks for your reviewing and advice